dialogue-annotations/src/test/java/com/palantir/dialogue/annotations/ParameterSerializerTest.java

```diff
@@ -0,0 +1,146 @@
+
+public final class ParameterSerializerTest {
+
+    private static final ParameterSerializer PLAIN = DefaultParameterSerializer.INSTANCE;
+
+    @Test
+    public void testSerializeBearerToken() {
+        BearerToken in = BearerToken.valueOf("token");
+        String out = "token";
+        assertThat(PLAIN.serializeBearerToken(in)).isEqualTo(out);
+    }
+
+    @Test
+    public void testSerializeAuthHeader() {
+        AuthHeader in = AuthHeader.of(BearerToken.valueOf("token"));
+        String out = "Bearer token";
+        assertThat(PLAIN.serializeAuthHeader(in)).isEqualTo(out);
+    }
+
+    @Test
+    public void testSerializeBoolean() {
+        for (Pair<Boolean, String> inOut : Arrays.asList(Pair.of(true, "true"), Pair.of(false, "false"))) {
+            boolean in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeBoolean(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testSerializeDateTime() {
+        OffsetDateTime in = OffsetDateTime.parse("2018-07-19T08:11:21+00:00");
+        String out = "2018-07-19T08:11:21Z";
+        assertThat(PLAIN.serializeDateTime(in)).isEqualTo(out);
+    }
+
+    @Test
+    public void testSerializeDouble() {
+        for (Pair<Double, String> inOut : Arrays.asList(Pair.of(1.234, "1.234"), Pair.of(1.2340, "1.234"))) {
+            double in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeDouble(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testSerializeFloat() {
+        for (Pair<Float, String> inOut : Arrays.asList(Pair.of(1.234f, "1.234"), Pair.of(1.2340f, "1.234"))) {
+            float in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeFloat(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testSerializeInteger() {
+        for (Pair<Integer, String> inOut : Arrays.asList(Pair.of(42, "42"), Pair.of(-42, "-42"))) {
+            int in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeInteger(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testSerializeLong() {
+        for (Pair<Long, String> inOut : Arrays.asList(Pair.of(42L, "42"), Pair.of(-42L, "-42"))) {
+            long in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeLong(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testChar() {
+        for (Pair<Character, String> inOut : Arrays.asList(Pair.of('c', "c"), Pair.of('\n', "\n"))) {
+            char in = inOut.getLeft();
+            String out = inOut.getRight();
+            assertThat(PLAIN.serializeChar(in)).isEqualTo(out);
+        }
+    }
+
+    @Test
+    public void testByte() {
+        for (Pair<Byte, String> inOut : Arrays.asList(Pair.of((byte) 0x0, "0"), Pair.of((byte) 0xde, "-34"))) {
```

Comment:
I am a bit torn on the byte here... @carterkozak what do you think?

Reply:
I might even consider leaving them out entirely, using the `int` method, and relying on implicit casting.

Reply:
you mean like force people to (int) their bytes? yeah, I meant to leave off bytes basically.

Reply:
Removed entirely.